app.js

```diff
@@ -85,10 +85,11 @@ router.post('/account', ratelimit({
 });
 
 
-const { findAccountsByPublicKey, findStakingDeposits } = require('./middleware/indexer');
+const { findAccountsByPublicKey, findStakingDeposits, findAccountActivity } = require('./middleware/indexer');
 
 router.get('/publicKey/:publicKey/accounts', findAccountsByPublicKey);
 router.get('/staking-deposits/:accountId', findStakingDeposits);
+router.get('/account-activity/:accountId/:offset?/:limit?', findAccountActivity);
```

Comment:
nit: I think from REST POV kinda makes sense to structure endpoints more like:

```
/account/:accountId/staking-deposits
/account/:accountId/activity
```

stuff like `offset` / `limit` should better go into query parameters

also given that we likely would want to get latest transactions first and then maybe load older transactions (and new might be appearing in between) – it's best to not rely on `offset`. Best to have parameters to specify block timestamp range, so that it's possible to implement pagination resistant to new data coming in.